examples/basic_ibe/backend/src/types.rs

```diff
@@ -0,0 +1,104 @@
+use candid::{CandidType, Principal};
+use ic_cdk::api::management_canister::main::CanisterId;
+use ic_stable_structures::{storable::Bound, Storable};
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+use std::borrow::Cow;
+
+pub const MAX_MESSAGES_PER_INBOX: usize = 10;
+
+#[derive(CandidType, Serialize, Deserialize, Clone, Debug)]
+pub struct Message {
+    pub sender: Principal,
+    pub receiver: Principal,
```

Comment:
IIUC the receiver could be sent once as part of the `Inbox`, since all messages in an inbox are associated with that single receiver.